src/infer_payload.h

```diff
@@ -70,6 +76,10 @@ class InferPayload : public std::enable_shared_from_this<InferPayload> {
   bool is_promise_set_;
   std::function<void(std::unique_ptr<InferResponse>)> callback_;
   std::shared_ptr<ResponseAllocatorUserp> response_alloc_userp_;
+  std::mutex request_deletion_mutex_;
+  bool is_request_deleted_;
+  intptr_t request_address_;
```

Comment:
Do you think we can combine the two variables? For example, when `request_address_ == 0L` means this object is not managing a request which includes the case after the request is deleted. In other words, we can set `request_address_ = 0L` when deleting the request.

The only caveat is do we need the information that this object was not initialized/set with a request and the request is "deleted", in the future.